devicesetup/src/main/res/layout/activity_success.xml

```diff
@@ -38,14 +38,29 @@
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:layout_gravity="center"
-            android:layout_marginBottom="24dp"
+            android:layout_marginBottom="12dp"
             android:layout_marginLeft="24dp"
             android:layout_marginRight="24dp"
             android:layout_marginTop="6dp"
             android:gravity="center"
             android:text="@string/setup_success_details"/>
 
         <TextView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="6dp"
+            android:text="@string/device_naming_label"/>
+
+        <EditText
```

Comment:
Seems like this would be a good place for [TextInputLayout](https://developer.android.com/reference/android/support/design/widget/TextInputLayout.html)